### PR TITLE
adding docstrings for ast module

### PR DESCRIPTION
**Pull request by dnabanita7**
## Description
I have added docstrings for the following classes in AST module.

- [x] attribute.py
- [x] callable.py
- [x] enum.py
- [x] globals.py
- [x] __init__.py
- [x] util.py
- [x] static_attr.py
- [x] module.py
- [x] property.py
I will complete writing docstrings for the remaining in the [spreadsheet](https://docs.google.com/spreadsheets/d/1ExD1EQIQtLx03a_h7lBPlvasVsmb38S02Bfw5a7Q62Y/edit#gid=0). Related to issue #5279 

## Affected Dependencies
NA

## How has this been tested?
pre-commit run --all-files

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by dnabanita7** [comment]
I will revert the changes made in ``klass.py``.

File: src/syft/ast/klass.py [hunk 1]
```diff
@@ -294,6 +294,16 @@ def __init__(
         return_type_name: Optional[str],
         client: Optional[Any],
     ):
+        """
```
Comment (tudorcebere) [inline src/syft/ast/klass.py:297]: Hi @dnabanita7!

This is not reflecting what the current implementation of Class is doing. If you want, we can setup a call to explain you the overall idea. Could you leave the docstring for Class to a separate PR? :)

File: src/syft/ast/enum.py [hunk 2]
```diff
@@ -20,14 +24,13 @@ def __init__(
         client: Optional[AbstractNodeClient] = None,
     ) -> None:
         """
-        An EnumAttribute represents the attributes of a Python Enum. Due to it's
-        constraints, they are only gettable, not settable.
+        Base constructor for Enum Attribute.
```
Comment (tudorcebere) [inline src/syft/ast/enum.py:27]: Not necessary, we can remove this.

File: src/syft/ast/globals.py [hunk 3]
```diff
@@ -29,6 +29,17 @@ def __init__(
         path_and_name: Optional[str] = None,
         return_type_name: Optional[str] = None,
     ):
+        """
```
Comment (tudorcebere) [inline src/syft/ast/globals.py:32]: Nice job! :fire: :fire:

**Review by tudorcebere — COMMENTED** [review]
Really nice job @dnabanita7! Left a few comments and requested a review from @jmaunon  and @IrinaMBejan. :+1:

**Review by jmaunon — DISMISSED** [review]
Hi @dnabanita7 
Thanks for your effort!! I think you have started for one of the most complex modules, great job!

The quality of the docstring is measured in the following way:

* `interrogate -P -ignore-private src/syft/ast: Returns the percentage of docstring coverage: `RESULT: FAILED (minimum: 80.0%, actual: 29.3%)`
* `pydocstyle --convention=google  src/syft/ast`: Raises errors according to Google convention (see [this link](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for examples)
* `darglint  src/syft/ast`: Raises errors according to docstring consistency

They can be installed simply by copy-paste the following command: `pip install interrogate pydocstyle darglint flake8`

This can be a bit a annoying, but following this criteria the QA will be improved a lot! Do not hesitate to write me if you have any question. We plan to new issues for the rest of modules.

File: src/syft/ast/klass.py (hunk 1 above)
Comment (dnabanita7) [reply src/syft/ast/klass.py:297]: Actually, I have revert those chances in klass.py now. But yes I am all in for a call to understand the working.

**Comment by dnabanita7** [comment]
> Hi @dnabanita7
> Thanks for your effort!! I think you have started for one of the most complex modules, great job!
> 
> The quality of the docstring is measured in the following way:
> 
> * `interrogate -P -ignore-private src/syft/ast: Returns the percentage of docstring coverage: `RESULT: FAILED (minimum: 80.0%, actual: 29.3%)`
> * `pydocstyle --convention=google  src/syft/ast`: Raises errors according to Google convention (see [this link](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for examples)
> * `darglint  src/syft/ast`: Raises errors according to docstring consistency
> 
> They can be installed simply by copy-paste the following command: `pip install interrogate pydocstyle darglint flake8`
> 
> This can be a bit a annoying, but following this criteria the QA will be improved a lot! Do not hesitate to write me if you have any question. We plan to new issues for the rest of modules.

Let me run these and get back to you. Thank you!

File: src/syft/ast/enum.py (hunk 2 above)
Comment (dnabanita7) [reply src/syft/ast/enum.py:27]: Alright!

**Comment by dnabanita7** [comment]
I have resolved most of the issues in the file ``enum.py`` and  ``globals.py``. The report on ``enum.py`` is as follows:
```
(nabanita07) (base) nabanita07@nabanita07:~/PySyft/src/syft/ast$ interrogate -P -ignore-private enum.py
RESULT: PASSED (minimum: 80.0%, actual: 83.3%)
Generated badge to /home/nabanita07/PySyft/src/syft/ast/nore-private
(nabanita07) (base) nabanita07@nabanita07:~/PySyft/src/syft/ast$ pydocstyle --convention=google enum.py
enum.py:1 at module level:
        D100: Missing docstring in public module
(nabanita07) (base) nabanita07@nabanita07:~/PySyft/src/syft/ast$ darglint enum.py
enum.py:__call__:91: DAR101: - **kwargs
enum.py:__call__:91: DAR101: - *args
enum.py:add_path:101: DAR101: - **kwargs
enum.py:add_path:101: DAR101: - *args
```
The report on ``globals.py`` are as follows:
```
(nabanita07) (base) nabanita07@nabanita07:~/PySyft/src/syft/ast$ interrogate -P -ignore-private globals.py
RESULT: FAILED (minimum: 80.0%, actual: 66.7%)
Generated badge to /home/nabanita07/PySyft/src/syft/ast/nore-private
(nabanita07) (base) nabanita07@nabanita07:~/PySyft/src/syft/ast$ pydocstyle --convention=google globals.py
globals.py:1 at module level:
        D100: Missing docstring in public module
globals.py:132 in public method `apply_node_changes`:
        D102: Missing docstring in public method
(nabanita07) (base) nabanita07@nabanita07:~/PySyft/src/syft/ast$ darglint globals.py
```

**Comment by dnabanita7** [comment]
I have added docstrings and inline comments wherever required in the `src/syft/ast` module except for `klass.py`. I am thinking of creating an another PR for it. All other methods of `syft/ast` have dosctrings. Some still fail the pydocstyle and darglint.

File: src/syft/ast/util.py [hunk 4]
```diff
@@ -1,14 +1,10 @@
 # -*- coding: utf-8 -*-
-"""
```
Comment (tudorcebere) [inline src/syft/ast/util.py:2]: Woah, this the definition of legacy. This can go away for sure

Comment (dnabanita7) [reply src/syft/ast/util.py:2]: Yeah!

**Review by tudorcebere — APPROVED** [review]
